components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -173,7 +171,12 @@ handler.on(PanelActions.signMessage.getType(), async (store: Store, payload: Sig
 handler.on(PanelActions.signMessageProcessed.getType(), async (store: Store, payload: SignMessageProcessedPayload) => {
   const apiProxy = await getAPIProxy()
   const braveWalletService = apiProxy.braveWalletService
-  braveWalletService.notifySignMessageRequestProcessed(payload.approved, payload.id)
+  await braveWalletService.notifySignMessageRequestProcessed(payload.approved, payload.id)
+  const signMessageRequest = await getPendingSignMessageRequest()
```

Comment:
why do we have to dispatch requests here?

Reply:
After a `signMessageProcessed` has occurred it will call `getPendingSignMessageRequest` and if there is still pending requests, it will call `signMessage` which will now update the list of pending requests and keep the panel open.

Reply:
I see, you will also need to do that here
https://github.com/brave/brave-core/pull/10811/files#diff-54c80cca70af79cdc89fd074c1f87ede5ce5600ca72f74f5f0f0db4b203981a6R198

all requests including hardware share the same queue

Reply:
ah yes, you are right! Good call, this has been updated!